dask_match/datasets.py

```diff
@@ -41,25 +43,29 @@ def _meta(self):
             "2000", "2000", dtypes, list(dtypes.keys()), "1H", states, self.kwargs
         )
 
+    @functools.lru_cache
```

Comment:
This is already cached on `divisions`.  Maybe use that instead?

Reply:
We cannot use `self.divisions` within `self.random_state` or `self._task`, because we need access to the original divisions (before `_partitions` was taken into account).  `self.divisions` will have already selected the `_partitions` indices.

Reply:
Makes sense.  Thanks!